lib/puppet/application/apply.rb

```diff
@@ -333,11 +333,11 @@ def setup
   private
 
   def read_catalog(text)
+    format = Puppet::Resource::Catalog.default_format
     begin
-      catalog = Puppet::Resource::Catalog.convert_from(Puppet::Resource::Catalog.default_format,text)
-      catalog = Puppet::Resource::Catalog.pson_create(catalog) unless catalog.is_a?(Puppet::Resource::Catalog)
+      catalog = Puppet::Resource::Catalog.convert_from(format, text)
     rescue => detail
-      raise Puppet::Error, "Could not deserialize catalog from pson: #{detail}", detail.backtrace
+      raise Puppet::Error, "Could not deserialize catalog from #{format}: #{detail}", detail.backtrace
```

Comment:
@Magisus has a separate PR open that wraps this string, https://github.com/puppetlabs/puppet/pull/5570 - could you update to follow that format? We're going to have conflicts either way depending on which is merged first